what4/src/What4/Expr/Builder.hs

```diff
@@ -2913,17 +2948,20 @@ instance IsExprBuilder (ExprBuilder t st fs) where
       Just LeqProof <- return $ isPosNat w
       bvUnary sym $ UnaryBV.uext u w
 
-    | Just (BaseIte _ _ c a b) <- asApp x
-    , Just a_bv <- asBV a
-    , Just b_bv <- asBV b = do
-      Just LeqProof <- return $ isPosNat w
-      a' <- bvLit sym w $ BV.zext w a_bv
-      b' <- bvLit sym w $ BV.zext w b_bv
-      bvIte sym c a' b'
-
     | otherwise = do
-      Just LeqProof <- return $ testLeq (knownNat :: NatRepr 1) w
-      sbMakeExpr sym $ BVZext w x
+      pmo <- CFG.getOpt (sbPushMuxOps sym)
```

Comment:
Quick check: looks like you had to do this in 6-8 other locations as well.  Do you think it's reasonable to float this `getOpt` to the top before the primary case on expr so that the case pattern match branches can be largely as they were with just the `pmo` check?  I'm fine either way (based on the assumption that `getOpt` is inexpensive), but just wanted to call this out for you to make a decision on.

Reply:
If I understand you correctly, you are suggesting to call `getOpt` at the beginning of each method's implementation to minimize the diff to the actual logic of the code itself? If so, I'm not sure that that would actually have the desired effect. In particular, each method of this `IsExprBuilder` instance is implemented using pattern guards, so in order to call `getOpt` before the pattern guards, we'd likely need to do something like this:

```hs
  bvZext sym w x = do
    pmo <- CFG.getOpt (sbPushMuxOps sym)
    if | Just xv <- asBV x = do ...
       | Just (BVZext _ y) <- asApp x = do ...
       |  ...
```

But using `MultiWayIf` (or a similar refactoring) here would cause the diff to be just as large (if not larger) than it is currently. As such, I opted to localize the `getOpt` calls to the parts of the code that actually make use of the option.